agent/consul/state/config_entry_intention.go

```diff
@@ -288,10 +288,13 @@ func readSourceIntentionsFromConfigEntriesForServiceTxn(
 		entry := v.(*structs.ServiceIntentionsConfigEntry)
 		for _, src := range entry.Sources {
 			if src.SourceServiceName() == sn {
+				kind := structs.GatewayServiceKindService
 				entMeta := entry.DestinationServiceName().EnterpriseMeta
-				kind, err := GatewayServiceKind(tx, entry.DestinationServiceName().Name, &entMeta)
-				if err != nil {
-					return nil, err
+				if entMeta.NamespaceOrDefault() != acl.WildcardName && entMeta.PartitionOrDefault() != acl.WildcardName {
+					kind, err = GatewayServiceKind(tx, entry.DestinationServiceName().Name, &entMeta)
+					if err != nil {
+						return nil, err
+					}
 				}
```

Comment:
Can these be assigned outside of the for-loop? So we don't need to query per source